src/Types.jl

```diff
@@ -511,7 +511,9 @@ function handle_repos_develop!(ctx::Context, pkgs::AbstractVector{PackageSpec},
                     # Relative paths are given relative pwd() so we
                     # translate that to be relative the project instead.
                     # `realpath` is needed to expand symlinks before taking the relative path.
-                    pkg.path = relpath(realpath(abspath(pkg.repo.url)), realpath(dirname(ctx.env.project_file)))
+                    project_dir = dirname(ctx.env.project_file)
+                    isdir(project_dir) && (project_dir = realpath(project_dir))
```

Comment:
This needs to call `realpath` on the part of `ctx.env.project_file` which exists, i.e. given
`/a/b/c/Project.toml`, if only `/a/b` exists, we need something like `joinpath(realpath(/a/b), c/Project.toml)`.

Reply:
Basically we need a `safe_realpath`, somehting like
```
julia> function safe_realpath(path)
           ispath(path) && return realpath(path)
           a, b = splitdir(path)
           return joinpath(safe_realpath(a), b)
       end
safe_realpath (generic function with 1 method)

julia> str = "/home/fredrik/Foo/test/Project.toml"
"/home/fredrik/Foo/test/Project.toml"

julia> realpath(str)
ERROR: SystemError: realpath: No such file or directory
Stacktrace:
 [1] #systemerror#39(::Nothing, ::Function, ::Symbol, ::Bool) at ./error.jl:106
 [2] systemerror at ./error.jl:106 [inlined]
 [3] realpath(::String) at ./path.jl:322
 [4] top-level scope at none:0

julia> safe_realpath(str)
"/home/fredrik/Foo/test/Project.toml"
```